Primo.js

```diff
@@ -213,7 +213,13 @@ function importPNX(text, url) {
 	
 	var itemType = ZU.xpathText(doc, '//display/type');
 	if(!itemType) {
-		throw new Error('Could not locate item type');
+		if ( ZU.xpathText(doc, '//facets/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//facets/rsrctype');
+		} else if ( ZU.xpathText(doc, '//search/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//search/rsrctype');
```

Comment:
You can write such code more compact by using the OR operator `||`,  i.e.
```js
var itemType = ZU.xpathText(doc, '//display/type') || ZU.xpathText(doc, '//facets/rsrctype') || ZU.xpathText(doc, '//search/rsrctype')
```
(Maybe split into several lines for readability.) Then the variable `itemType` equals to the first expression, it this is not `null`, or otherwise equals to the second expression, if this is not `null`, or otherwise equals to the third expression.